internal/libs/autofile/autofile.go

```diff
@@ -83,33 +84,68 @@ func OpenAutoFile(path string) (*AutoFile, error) {
 		}
 	}()
 
-	go af.closeFileRoutine()
+	go af.syncFileRoutine()
 
 	return af, nil
 }
 
 // Close shuts down the closing goroutine, SIGHUP handler and closes the
 // AutoFile.
 func (af *AutoFile) Close() error {
-	af.closeTicker.Stop()
+	af.refreshTicker.Stop()
 	close(af.closeTickerStopc)
 	if af.hupc != nil {
 		close(af.hupc)
 	}
 	return af.closeFile()
 }
 
-func (af *AutoFile) closeFileRoutine() {
+func (af *AutoFile) syncFileRoutine() {
 	for {
 		select {
-		case <-af.closeTicker.C:
-			_ = af.closeFile()
+		case <-af.refreshTicker.C:
+			_ = af.syncFile()
 		case <-af.closeTickerStopc:
 			return
 		}
 	}
 }
 
+func (af *AutoFile) syncFile() (err error) {
```

Comment:
I don't see any particular reason that we need to declare the `err` variable in the method signature. 

also this implementation never returns an error (but perhaps should. 

is it possible to write a test for this?

Reply:
>I don't see any particular reason that we need to declare the err variable in the method signature.

I was using it before, but I'll remove it, thanks.

>also this implementation never returns an error (but perhaps should.

I panicked to expose unhandled and rare errors given that the routines that previously invoked .closeFile ignored the error. But sure I'll return the error instead.

>is it possible to write a test for this?

Yes, please.

Reply:
can you add this test?

Reply:
I did add it. Please take a look at the new test in there.

Reply:
Actually this test is going to be almost impossible to write with what seems to be a bug in Posix systems that I've reproduced and isolated at https://twitter.com/odeke_et/status/1468036751952207872 :-(

Reply:
I don't think that is a bug. Generally speaking, POSIX permissions are checked when the resource is acquired, and once the descriptor is assigned you generally keep those permissions until it's closed. Similarly, unlinking the file does not make it unwritable; the file data remain pinned until all open file descriptors on that inode close, and then it is released.

I don't know if Linux handles it differently, but this behaviour isn't particularly surprising.

Reply:
@creachadair well, it is a design flaw in *NIX and a bug. It being accepted for decades doesn't mean it isn't a bug :-) As I've clearly exhibited, we can't even make assumptions about permissions being propagated. There is some discourse on my tweet about it https://twitter.com/odeke_et/status/1468036751952207872?s=21 I have a fix for it but it very tangential from work on Cosmos and requires kernel patches to the different systems, but I'll publish it and a blog post too :-) Thanks for the chiming in @creachadair

Reply:
It's reasonable to argue against the design, certainly. But consider the alternative: If changing the permission bits on an open file could invalidate an already-open filehandle, it would have very unpredictable and difficult-to-debug effects on running processes. If the concern is security vulnerabilities, that would I think be at least as bad.

It does require caution, and it may be more subtle than it should be—but I think it was a fairly reasonable tradeoff given the alternatives.